src/components/layout/CardHeader.tsx

```diff
@@ -1,26 +1,32 @@
 import classnames from 'classnames';
+import type { JSX } from 'preact';
 
+import type { PresentationalProps } from '../../types';
 import { downcastRef } from '../../util/typing';
 
 import { CancelIcon } from '../icons';
 import { IconButton } from '../input';
 
 import CardTitle from './CardTitle';
 
-/**
- * @typedef {import('../../types').PresentationalProps} CommonProps
- * @typedef {import('preact').JSX.HTMLAttributes<HTMLElement>} HTMLAttributes
- *
- * @typedef CardHeaderProps
- * @prop {string} [title]
- * @prop {() => void} [onClose] - Optional callback to close the Card. When
- *   present, a close button will be rendered
- */
+type ComponentProps = {
+  title?: string;
+
+  /**
+   * Optional callback for close-button click. When present, a close button
+   * will be rendered.
+   */
+  onClose?: () => void;
+};
+
+type HTMLAttributes = JSX.HTMLAttributes<HTMLElement>;
+
+export type CardHeaderProps = PresentationalProps &
+  ComponentProps &
+  HTMLAttributes;
```

Comment:
It's very early days for TypeScript conversion of these package components, so I imagine we'll feel our way a bit for the best (clearest) way to express component prop types. We can express `CardHeaderProps` without defining `ComponentProps` and `HTMLAttributes` types separately, but I felt like this was a little easier to read.